### ISSUE TITLE
Question: CUDNNv4 SpatialBatchNorm momentum

### ISSUE THREAD

**Issue by gcr** [soumith/cudnn.torch#91]
I noticed that `cudnn.h` includes this rather cryptic comment for the `cudnnBatchNormalizationForwardTraining()` function:

```
/* MUST use factor=1 in the very first call of a complete training cycle.
    Use a factor=1/(1+n) at N-th call to the function to get
    Cumulative Moving Average (CMA) behavior
    CMA[n] = (x[1]+...+x[n])/n
    Since CMA[n+1] = (n*CMA[n]+x[n+1])/(n+1) =
    ((n+1)*CMA[n]-CMA[n])/(n+1) + x[n+1]/(n+1) =
    CMA[n]*(1-1/(n+1)) + x[n+1]*1/(n+1) */
double                              exponentialAverageFactor,
```

In the current version of the code, this is just the `momentum` parameter which defaults to 0.1. Is it worth implementing the proper averaging behavior? The advantage is that this would allow us to capture the mean and variance across the entire training set.

**Comment by soumith** [comment]
It's not worth it. The proper averaging behavior on training set has to be computed freezing training, and takes way too long, and there's no accuracy gain for doing so
